### ISSUE TITLE
Call to a member function toJSON() on null

### ISSUE THREAD

**Issue by MagicLegend** [propelorm/Propel2#1666]
_This is referring to #1426 ; given that it is the same error._

However, I am not getting how to solve it. Why does the `toJSON()` function not simply accept null, and return an empty object / null? How am I supposed to properly handle this situation? I explicitly do _not_ want to use `findById()` that returns a collection. It would also not be preferred to have to build manual checks if the given ID is present in the database, this hassle is exactly why I want to use an ORM :-)

Why doesn't this just work?
```php
BookQuery::create()->findPk(1000)->toJSON();
```

```
PHP Fatal error:  Uncaught Error: Call to a member function toJSON() on null in /home/magiclegend/Documents/test/src/api/book.php:41
Stack trace:
#0 {main}
  thrown in /home/magiclegend/Documents/test/src/api/book.php on line 41
```

Am I supposed to add an extra function call to catch the exception? From my Node background I assumed that surrounding it with a `try catch` would solve the issue, but it does not seem to want to catch. What is the proper way to go here?

**Comment by dereuromark** [comment]
Any method call on null is not possible.
So you should first read it and check !== null before doing operations on it.

Another idea could be use a findOrFail() method if that exists for your case.

**Comment by MagicLegend** [comment]
> Any method call on null is not possible.

Yeah, which is logical. I had just made a [rookie mistake](https://stackoverflow.com/questions/2172715/try-catch-block-in-php-not-catching-exception#comment20035245_7892917) in the try-catch structure I tried; meaning it wouldn't catch the exception. I have put the correct one around it now; and it works like I expected it to. (Catching if an invalid Pk is given)

> So you should first read it and check !== null before doing operations on it.

Well, as I said; I hoped to keep it a one-liner. Of course doing the manual null check works, but I was hoping that Propel would handle that for me, or I could implicitly test for it with a try-catch structure.

A try-catch around it works now. Thank you anyhow for taking the time to answer my question.
